Sources/ConfidenceProvider/ConfidenceFeatureProvider.swift

```diff
@@ -464,45 +548,42 @@ extension ConfidenceFeatureProvider {
                 session: session,
                 localOverrides: self.localOverrides.merging(localOverrides) { _, new in new },
                 flagApplier: flagApplier,
+                storage: storage,
                 cache: cache,
+                initializationStrategy: initializationStrategy,
                 applyStorage: applyStorage
             )
         }
 
         /// Creates the `ConfidenceFeatureProvider` according to the settings specified in the builder.
         public func build() -> ConfidenceFeatureProvider {
-            let flagApplier = flagApplier ?? FlagApplierWithRetries(
-                httpClient: NetworkClient(region: options.region),
-                storage: DefaultStorage(filePath: "applier.flags.cache"),
-                options: options
-            )
+            let flagApplier =
+                flagApplier
+                ?? FlagApplierWithRetries(
+                    httpClient: NetworkClient(region: options.region),
+                    storage: DefaultStorage(filePath: "applier.flags.cache"),
```

Comment:
why do we create a new storage instance here and not reuse the one in the builder?

Reply:
Oh, maybe I'm mixing up apply cache and flags cache?

Reply:
yes, this one is `applier.flags.cache`.